--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -104,8 +104,7 @@

 if(PNINEXUS_USE_BOOST_FILESYSTEM MATCHES ON)
   set(PNINEXUS_FILESYSTEM_LIB "boost_filesystem")
-  find_package(Boost 1.60 COMPONENTS system
-                                   regex
+  find_package(Boost 1.60 COMPONENTS regex
                                    date_time
                                    unit_test_framework
 				   filesystem
@@ -112,8 +111,7 @@
 				   REQUIRED)
 else()
   set(PNINEXUS_FILESYSTEM_LIB "stdc++fs")
-  find_package(Boost 1.60 COMPONENTS system
-                                   regex
+  find_package(Boost 1.60 COMPONENTS regex
                                    date_time
                                    unit_test_framework
 				   REQUIRED)
